sample-compose/app/src/main/java/co/nimblehq/sample/compose/model/UiModel.kt

```diff
@@ -6,7 +6,8 @@ import kotlinx.parcelize.Parcelize
 
 @Parcelize
 data class UiModel(
-    val id: String
+    val id: String = "",
+    val url: String = "https://www.google.com/"
```

Comment:
May I know any purpose for adding `url` in the model 🤔  I didn't see where we will use it only show in `ThirdScreen`, Can we have only `id`? 🙏 
```suggestion

```

Reply:
@Wadeewee It is to ensure we can navigate with a Parcelable data which include `url`. There are 2 ways to navigate with parcelable data. Using `savedInstanceState` and `navArgument`. We can serialize the whole model and use it inside `navArgument` too. In that case, we need to encode `url` before serializing the object. Because the navigation uses `nav route` and it somehow assume the `url` as part of the `nav route`.

Reply:
> It is to ensure that we can navigate with a Parcelable Object which include url

@kaungkhantsoe, Not sure If I get your point, but removing `url` it's still a `Parcelable Object` 🤔 

Here's my result when remove `url`:

https://user-images.githubusercontent.com/28002315/215727331-80ccea3f-3011-4d59-94ba-b55b354de3a0.mov

Please correct me If I missed something 🙏  cc @hoangnguyen92dn @Tuubz

Reply:
@Wadeewee Yes, it is working since we are using `savedInstanceState` for now. There is another way to navigate without using `savedInstanceState`. We haven't implemented it yet. I can remove it for now. I just want us to be aware of the danger of `url` here.

Reply:
`savedInstanceState` has a drawback as describe in [this comment](https://github.com/nimblehq/android-templates/pull/390#discussion_r1090170856).

Reply:
Got it. Since `url` has that kind of problem with navigation, we cannot be sure whether `savedInstanceState` might or might not have a problem later. I will remove it for now. cc @Tuubz @Wadeewee

Reply:
> In that case, we need to encode url before serializing the object. Because the navigation uses nav route and it somehow assume the url as part of the nav route.

@kaungkhantsoe Hold onnnn... With the `Third` App destination, we don't add any extra information to the nav route, it's simply `"third"` like `"home"`, the parcel object will be retained in the `savedInstanceState`, it doesn't related to the nav route.

Reply:
Btw, I am fine with the object containing `url` or not, the idea is passing the data object to destination 😄 @kaungkhantsoe @Wadeewee

Reply:
You can check out [this thread](https://stackoverflow.com/a/67133534) and its comments why we could have a problem with `url`. We haven't implemented the `serialization` way for now because we are worry that it might block the thread if we serialize a big object. Unlike using `savedInstanceState`, `serialization` way doesn't have a drawback as described in [this comment](https://github.com/nimblehq/android-templates/pull/390#discussion_r1090170856)  cc @hoangnguyen92dn @Wadeewee